### ISSUE TITLE
Create Onboarding Process for Discord

### ISSUE THREAD

**Issue by DanM3rcurius** [SecretDecks/Documentation#13]
## Goal
SecretDecks community offers an onboarding flow, where newcomers are being introduced to the community and vice versa. 
Some kind of community "database" is most likely needed, in order to acquaint one with the members (if it starts to grow).

## Specifications
1. Draft
2. Discuss
3. Document results
4. Test
5. Review
6. Discuss
7. Document results 
8. Repeat from 4.

**Comment by wlstylr** [comment]
## Desired outcome  
Onboarding into the SecretDecks community should be an intuitive and educational process that gives the new community member a clear understanding of SecretDecks’ values and mission, while allowing them to quickly understand how they can contribute.

**A rough draft of the process could be something similar to as follows:**

1. Candidate clicks on Discord invite link
2. When entering Discord, candidates are first presented with the Mindfulness Policy followed by the Charter.  It is required that candidates sign the MP and Charter before having the ability to participate in the Discord Server.  This can be accomplished through using a bot and setting certain roles on the server. 
3. After signing the MP and Charter the candidate achieves access to all server channels and is prompted to leave an introduction in the welcome or intros channel.
4. If an intro is left, an existing SD member should reply, welcome them to the server, and try to learn a bit more about how they want to participate. 
5. If it turns out they want to actively participate, the new SD member should be directed to the appropriate area of the server and an existing community member who can facilitate next steps.
6. Perhaps after a certain amount of time and contributions are recorded, the new SD member receives an NFT that represents completing onboarding process?



## Questions and challenges

1. If SD is to create a “database” of active community members where should this be hosted and how is it controlled?  
2. Should the database be hosted on IPFS?  Google docs? Airtable?  I would advise against comprising towards a centralized solution.  Once you make the choice to use an "easy" tool it becomes increasingly difficult to migrate to a tool that better aligns with community values as the community grows.
3. If there is a defined onboarding process, does there also need to be a defined offboarding process?  What would this look like?
4. Can anyone join the SD discord or does an existing member need to vouch for the new candidate?
5. Will Discord invite link be publicly available or only accessible through shared links from existing members?

**Comment by stephen-rowan** [comment]
Hi @wlstylr - it may be worthwhile considering some form of training or mentorship at "Desired outcome 4)" as this can be a point where new members may feel lost. This could take the form of on-boarding workshops or prefixed to SD weekly meetings? It is good to have some form of interaction in onboarding - so it does not feel just like a sign-up process?

The onboarding process should be summarized in SD's "contributions policy" (or linked from there). The standard open-source docs typically on a repo landing page are pictured below (with the README doing the job of a general intro) - 

![2021-05-26](https://user-images.githubusercontent.com/25156451/119696197-d8d8b100-be46-11eb-9b35-506c3a2cc05c.png)

"Questions and challenges 1)" - this is an ongoing issue in a few projects. I have seen a voluntary approach - where members post their details, profile, and availability in a public repo (e.g., csv) or Discord channel. Members who wish to be anon could a pseudonym. 

In the long-term this will need to be resolved by a DID system - because of ID and data privacy legislation in various countries.

**Comment by wlstylr** [comment]
@stephen-rowen ("Desired outcome 4)" I agree that on-boarding workshops would be valuable, especially as SD grows and garners increasing interest from the general Cardano/Catalyst community.  Another idea would be to create an onboarding video (just record and post one of the onboarding workshops).  I don't think that would replace an actual weekly workshop where folks can interact and ask questions, but it would provide something of reference until the workshop event. 

"Questions and challenges 1)" I agree that a public repository that allows pseudonyms should suffice for the short term.

**Comment by wlstylr** [comment]
@DanM3rcurius @stephen-rowan 
Not the most polished, but provides visuals...  Feel free to add comments

https://lucid.app/lucidchart/invitations/accept/inv_49662764-fbfd-4674-82de-e303b2c41612?viewport_loc=49%2C-25%2C1436%2C812%2C0_0

**Comment by DanM3rcurius** [comment]
> @DanM3rcurius @stephen-rowan
> Not the most polished, but provides visuals... Feel free to add comments
> 
> https://lucid.app/lucidchart/invitations/accept/inv_49662764-fbfd-4674-82de-e303b2c41612?viewport_loc=49%2C-25%2C1436%2C812%2C0_0

Hehe @wlstylr that looks grand! I didn't check if someone has been working on this task already and was doing [this](https://github.com/DanM3rcurius/DAO.governance/blob/main/SexretDecks-DiscordOnboarding.jpg)  just now 😂 
Can we combine them in one or the other? I love the level of detail of your diagram, but would like to keep the interim "Trainee" position depicted in my diagram.

**Comment by DanM3rcurius** [comment]
@wlstylr I also just saw that I didnt put me as assignee before starting the task, nor did I push it to the "doing" column in the kanban board 😅 Then I would've seen maybe that you did a comment and already started working on it^^ 
We should make a habit of this, I'll also add it to the github workflow text.

**Comment by wlstylr** [comment]
@DanM3rcurius I was just adding comments and it turned into adding a diagram as a comment.  I also wasn't taking a formalized approach (assigning myself, moving to "doing", etc).  I really like how you broke up the flow into #_core and #_shell, which provides clear guidelines on how to progress within the SD organization.   Feel free to incorporate what you like from my diagram and comments :)

**Comment by DanM3rcurius** [comment]
**What Discord bot to use for moderation/onboarding?**
We need him to greet and point new members to the rules/process; as well as ideally check if new member wrote in introductions-channel and give new role "Shell"

- [ ] [Arcane](https://top.gg/bot/437808476106784770)
- [ ] [Loritta](https://top.gg/bot/loritta)
- [ ] [ProBot](https://top.gg/bot/probot)
- [ ] a different one:

**Comment by wlstylr** [comment]
Not sure @DanM3rcurius ... So, I created a clone of the SD discord server here https://discord.gg/ZyWZZHzc9d that we can use to test the bots and onboarding flow.   If we have the invite link go directly to #welcome-and-rules do we still need to have the bot greet and point?

**Comment by tamagit80** [comment]
cool I'm going to test it later

**Comment by wlstylr** [comment]
Sounds good; I just added the Arcane bot and set it up to send a DM.  You should land directly in the #welcome-and-rules channel.  I think the two github links that are currently in the welcome channel message will need to be modified slightly so that the content is very easy to understand for someone new to github and SD.   

I haven't yet figure out how to automate the role change after a new person leaves an intro message.

**Comment by tamagit80** [comment]
I think I don't have permissions as admin.

**Comment by DanM3rcurius** [comment]
> Not sure @DanM3rcurius ... So, I created a clone of the SD discord server here https://discord.gg/ZyWZZHzc9d that we can use to test the bots and onboarding flow.

perfect! lets test em, I'll get on that now!

>  If we have the invite link go directly to #welcome-and-rules do we still need to have the bot greet and point?

yes and no. If we make the welcome statement so that everyone has to read it, and its clear and short then we don't need the greeting bot. Am I mistaken?

**Comment by DanM3rcurius** [comment]
I tried understanding the [commands here](https://arcanebot.xyz/commands?prefix=a) not sure I can be of help with this bot...  @wlstylr can you please give admin rights to @tamagit80 on the test server?

**Comment by wlstylr** [comment]
@DanM3rcurius I'll grant @tamagit80 admin rights the next time he is in the test server.  I'm also unfamiliar with this bot.  I was just trying it out because it was first on your list and seems to offer a lot of the functionality that we need.  I'm going to try watching some tutorials before giving up on it.

**Comment by DanM3rcurius** [comment]
@wlstylr been playing around with the "Community" Setting in Discord! Check it out in Server settings. it now produces a welcome screen. I need to setup another discord account to test it.. But it looks like the permission levels and roles are easier to define (and even automate on certain actions perhaps?). Could be that we don't even need a bot at this stage?

**Comment by wlstylr** [comment]
@DanM3rcurius Nice find!  I ended up going to the river with the family for the afternoon; checking out community settings now.  Also, if you set up a test server of your own, there is a quick way to create a template from the existing server as a starting point.  https://www.youtube.com/watch?v=y9E0kvbAhl4

**Comment by DanM3rcurius** [comment]
Sounds amazing, hope you had a joyous day!
Actually @tamagit80 found the community feature, I just activated it ^^
Already created a template, will post it here in a second so we can test away!

regards,
D.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by DanM3rcurius** [comment]
> ## Questions and challenges
> 1. If SD is to create a “database” of active community members where should this be hosted and how is it controlled?

I'd propose to use 3rd party tools/social media accounts that are added to the github account as notes (obligatory)
Until we have DID and a proper view on them on our website or sth else; we can take screenshots from the discord member view, like this:
<img width="708" alt="Screenshot 2021-06-14 at 13 59 32" src="https://user-images.githubusercontent.com/80257662/121888831-c0a6d400-cd18-11eb-925e-f7f6162e1db8.png">
Also "active" in that case means a contribution survey will be filled out. we then have somewhat of a directory of open data that we can use to make a nice view on the website. Low priority right now, or? @stephen-rowan @wlstylr 

> 2. Should the database be hosted on IPFS?  Google docs? Airtable?  I would advise against compromising towards a centralized solution.  Once you make the choice to use an "easy" tool it becomes increasingly difficult to migrate to a tool that better aligns with community values as the community grows.

yes, see above. let's keep the data near the member not near SD (except contribution survey), until we start playing around with DID

> 3. If there is a defined onboarding process, does there also need to be a defined offboarding process?  What would this look like?

I jsut finished setting up the basic discord onboarding. new members  have to verify mail, accept rules; then intro in channel, (pick their skills, optional)  and then head over to "1st quest" to read the mindfulness policy and agree by clicking; then they are shell members.
Should I create a tutorial video for this? otherwise I'd do the github video tutorial first (as I feel its needed more urgently)

> 4. Can anyone join the SD discord or does an existing member need to vouch for the new candidate?

Anyone can join, but needs to go through onboarding steps.

> 5. Will Discord invite link be publicly available or only accessible through shared links from existing members?

Will make it public after today, when I tested the onboarding a bit more. @SecretDecks/founding-core You're welcome to test with me by creating a new discord account (i know its a bit troublesome and the weather is so nice 😅)

Sorry again for being inactive last week!

**Comment by DanM3rcurius** [comment]
Basic onboarding works and has been deemed *"ok for now"* by current members: @DanM3rcurius , Jonathan Looney , @wlstylr
